### ISSUE TITLE
handle no-route errors

### ISSUE THREAD

**Issue by x6iae** [devcenter-square/disease-info#57]
we should add root route and prevent non-existent routes from throwing errors and breaking the app on production... 

maybe redirect non-existent routes to the 404.html inside public?

**Comment by luizvarela** [comment]
We can create a new strategy to handle the API errors, not only 404, but 422 or 401 or other HTTP errors.
as this is a JSON API we don't need to return html responses but just json response, something like this:

```
{  
   errors:{  
      resource:[  
         "Some explanation about the error"
      ]
   },
   data:{ 
   }
}
```

I can think how to handle this.

**Comment by AfolabiOlaoluwa** [comment]
Surely it will not be for only 404. I was thinking with @sunday to redirect to the HTML public document provided. But with this comment I agree we should do away with HTML. I will deal with it using rescue_from and simplify it to handle other errors as well.

**Comment by AfolabiOlaoluwa** [comment]
@luizvarela @SundayAdefila @osioke I have added root route [in this commit](https://github.com/devcenter-square/disease-info/commit/b20c22297298e85e4053cfacc2d7476be54de8a8).

**Comment by AfolabiOlaoluwa** [comment]
@SundayAdefila @luizvarela Solved. This is what I came up with [here](https://github.com/devcenter-square/disease-info/pull/58) also Route/diseases to root [in this commit](https://github.com/devcenter-square/disease-info/commit/b20c22297298e85e4053cfacc2d7476be54de8a8). However @luizvarela I beg of you to write test for this PR after merged.

**Comment by x6iae** [comment]
Fixed with #58
